workflows/virtual-screening-blueprint/definition/main.nf

```diff
@@ -0,0 +1,153 @@
+nextflow.enable.dsl = 2
+
+// static data files are in nextflow.config
+workflow {
+    //Convert to files
+    if (params.fasta_path[-1] == "/") {
+        fasta_path = params.fasta_path + "*"
+    } else {
+        fasta_path = params.fasta_path
+    }
+
+    if (params.smi_input_path[-1] == "/") {
+        smi_input_path = params.smi_input_path + "*"
+    } else {
+        smi_input_path = params.smi_input_path
+    }
+
+
+    sequences = Channel.fromPath(fasta_path)
+        .collectFile(name: 'temp.fasta')
+        .splitFasta(record: [id: true, seqString: true])
+        .map { record -> "${record.id}zzzz${record.seqString} " }
+        .collectFile(name: 'combined_seqs.txt')
+
+    RunAlphaFold2(
+        params.alphafold2_model,
+        params.alphafold2_script,
+        sequences
+    )
+
+    smiles = Channel.fromPath(smi_input_path)
+        .collectFile(name: 'smi_input.smi')
+
+    RunMolMIMGenerate(
+        params.molmim_script,
+        params.molmim_model,
+        smiles,
+        params.num_molecules
+    )
+
+    RunDiffdock(
+        params.diffdock_script,
+        params.diffdock_model,
+        RunMolMIMGenerate.out.sdfs,
+        RunAlphaFold2.out.pdbs,
+        params.num_poses
+    )
+}
+
+process RunAlphaFold2 {
+    label 'alphafold2'
+    errorStrategy 'retry'
+    cpus { 2 * Math.pow(2, task.attempt+2) }
```

Comment:
this looks off/too high. I think this is saying the minimum resources required here are 16 vCPUs and 64GB memory on (g5.4xlarge).

Reply:
for both cpus and memory this should remove the `+2`

Reply:
The recommended mem to run alphafold2 is 64GB, I think I got OOR using 16GB, let me try again for 32GB mem.

Reply:
got it - if that's what they recommend can stay with it